types: TextRequest class type is incorrect due to skipLibChecks: true

The type definition here:
https://github.com/Dash-Industry-Forum/dash.js/blob/4b0b16d6b031e418d279a5e4ba67c08d35022c0d/index.d.ts#L4107-L4114

generates a type error when `skipLibChecks: false` is added to a containing project, because this type is incorrect.

##### Environment
* typescript: ^5.1.3
* Dash.js version: v4.7.0

##### Steps to reproduce
1. checkout isolated reproduce case [here](https://github.com/flowplayer/bug-reports/tree/main/dash-types-4.7.0)
2. `cd dash-type-4.7.0` 
3.  run `yarn fail` convenience script to reproduce the case with `tsc`

##### Console output
```
yarn build
yarn run v1.22.17
$ tsc && rimraf dist && webpack --config build/webpack.prod.js
index.d.ts:4110:9 - error TS2416: Property 'url' in type 'TextRequest' is not assignable to the same property in base type 'FragmentRequest'.
  Type 'string | null' is not assignable to type 'string'.
    Type 'null' is not assignable to type 'string'.

4110         url: string | null;
             ~~~


Found 1 error in index.d.ts:4110
```

This is because in Typescript, you cannot change the type of an already declared property like this.  By default `tsc init` configures a project as `"skipLibCheck": true`

Which means that this silently fails as the types are [never checked in the dash.js `index.d.ts` file during compile time.](https://www.typescriptlang.org/tsconfig#skipLibCheck)

--- a/index.d.ts
+++ b/index.d.ts
@@ -4067,7 +4067,7 @@ declare namespace dashjs {
         startTime: number;
         timescale: number;
         type: 'InitializationSegment' | 'MediaSegment' | null;
-        url: string;
+        url: string | null;
         wallStartTime: number | null;
     }
 

--- a/package.json
+++ b/package.json
@@ -50,7 +50,7 @@
         "sinon": "^7.3.2",
         "stream-browserify": "^3.0.0",
         "string-replace-loader": "^3.0.1",
-        "typescript": "^4.1.5",
+        "typescript": "^5.1.3",
         "webpack": "^5.75.0",
         "webpack-cli": "^5.0.1",
         "webpack-dev-server": "^4.11.1",

--- a/tsconfig.json
+++ b/tsconfig.json
@@ -3,6 +3,9 @@
     "index.d.ts"
   ],
   "compilerOptions": {
-    "noImplicitAny": true
+    "noImplicitAny": true,
+    "strict": true,                                      /* Enable all strict type-checking options. */
+    "skipLibCheck": false
   }
+  
 }

